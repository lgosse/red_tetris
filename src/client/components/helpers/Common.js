import React from "react";

import styled from "styled-components";
import global from "../../styles/global";

export const Spacer = styled.div`
  flex: 1;
`;

export const Button = styled.button`
  padding: 15px 25px;
  font-family: ${global.font.family.game};
  transition: all 0.5s;
  border: none;
  background-color: unset;
  outline: none;
  cursor: pointer;

  ${props => props.width !== undefined && `width: ${props.width}`};
  ${props => props.margin !== undefined && `margin: ${props.margin}`};
  ${props => props.marginTop !== undefined && `margin-top: ${props.marginTop}`};
  ${props =>
    props.marginBottom !== undefined && `margin-bottom: ${props.marginBottom}`};
  ${props =>
    props.marginLeft !== undefined && `margin-left: ${props.marginLeft}`};
  ${props =>
    props.marginRight !== undefined && `margin-right: ${props.marginRight}`};
  ${props =>
    props.primary
      ? `
          background-color: ${global.color.primary};
          border: 1px solid ${global.color.accent};
          color: ${global.color.accent};

          &:hover {
            background-color: ${global.color.accent};
            color: ${global.color.primary};
          }
        `
      : `
          background-color: ${global.color.accent};
          color: ${global.color.primary};

          &:hover {
            background-color: ${global.color.accent};
            color: ${global.color.accent};
          }
    `};
`;

export const FlexContainer = styled.div`
  display: flex;
  ${props => props.flex && `flex: 1`};
`;

export const FlexSpacer = styled.div`
  flex: 1;
`;

export const FullSizeContainer = styled.div`
  width: 100%;
  ${props => props.padding && `padding: ${props.padding}`};
  ${props => props.flexContainer && `display: flex`};
  ${props => props.flex && `flex: 1`};
`;

const Hexagone = styled.span`
  display: block;
  position: relative;
  width: 20px;
  height: 11.54700538px;
  background-color: ${props => (props.primary ? `#bb3c2f` : `#dcddd8`)};

  margin: 6.66666667px 0;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);

  &:before,
  &:after {
    content: "";
    position: absolute;
    width: 0;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
  }

  &:before {
    bottom: 100%;
    border-bottom: 6.66666667px solid
      ${props => (props.primary ? `#bb3c2f` : `#dcddd8`)};
  }

  &:after {
    top: 100%;
    width: 0;
    border-top: 6.66666667px solid
      ${props => (props.primary ? `#bb3c2f` : `#dcddd8`)};
  }
`;

const HexaContainer = styled.div`
  height: 6px;
`;

export const Paragraph = styled.div`
  ${props => {
    if (props.width && props.center) {
      return `margin: auto; width: ${props.width}; text-align: center`;
    } else if (props.width && !props.center) {
      return `width: ${props.width}`;
    } else if (!props.width && props.center) {
      return `text-align: center`;
    }
  }};
  ${props => props.color && `color: ${global.color[props.color]}`};
  font-size: ${props => (props.size ? props.size : `14px`)};
  ${props => props.padding && `padding: ${props.size}`};
  ${props => props.gameFont && `font-family: ${global.font.family.game}`};
`;

export const RedContainer = FullSizeContainer.extend`
  background-color: ${global.color.primary};
  background-image: ${global.assets.backgroundRed};
`;

export const LightContainer = FullSizeContainer.extend`
  background-image: ${global.assets.backgroundLight};
  background-color: ${global.color.accent};
`;

export const GameFont = styled.span`
  font-family: ${global.font.family.game};
`;

export const Input = styled.input`
  box-sizing: border-box;
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  outline: none;
  display: block;
  width: 234px;
  padding: 7px;
  border: none;
  border-bottom: 1px solid #ddd;
  background: transparent;
  margin-bottom: 10px;
  font: 16px Arial, Helvetica, sans-serif;
  height: 45px;
  color: ${global.color.accent};

  &::placeholder {
    color: ${global.color.accent};
  }
`;

export const HexaSeparator = props => (
  <HexaContainer>
    <Hexagone primary={props.primary} />
  </HexaContainer>
);

const FaIcon = props => <i className={`fa fa-${props.className}`} />;

export const Icon = styled(FaIcon)`
  ${props => props.primary && `color: ${global.color.primary}`};
  ${props => props.accent && `color: ${global.color.accent}`};
  ${props => props.size && `font-size: ${props.size}`};
`;
