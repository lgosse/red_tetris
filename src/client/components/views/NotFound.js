import React from "react";
import styled from "styled-components";
import global from "../../styles/global";

const Code404 = styled.div`
  color: ${global.color.primary};
  font-family: ${global.font.family.game};
  font-size: 40px;
  text-align: center;
  width: 100%;
`;

const Message = styled.div`
  text-align: center;
  font-size: ${global.font.size.subtitle};
  color: ${global.color.primary};
  width: 300px;
  margin: auto;
`;

const containerStyle = {
  width: "100%"
};

const NotFound = () => {
  return (
    <div style={containerStyle}>
      <Code404>404</Code404>
      <Message>We're sorry but your requested page is unknown to us :(</Message>
    </div>
  );
};

export { containerStyle, Code404, Message };

export default NotFound;
