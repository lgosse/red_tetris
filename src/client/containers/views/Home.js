import React from "react";
import { withRouter } from "react-router-dom";
import RedTetrisHeader from "../../components/home/RedTetrisHeader";
import RedTetrisPreview from "../../components/home/RedTetrisPreview";
import BeginAdventure from "../../components/home/BeginAdventure";
import {
  FullSizeContainer,
  HexaSeparator
} from "../../components/helpers/Common";

const Home = () => {
  return (
    <FullSizeContainer>
      <RedTetrisHeader />
      <HexaSeparator />
      <RedTetrisPreview />
      <BeginAdventure />
    </FullSizeContainer>
  );
};

export default Home;
