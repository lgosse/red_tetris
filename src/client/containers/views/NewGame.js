import React from "react";
import styled from "styled-components";
import { connect } from "react-redux";

import global from "../../styles/global";
import { newGame, joinGame } from "../../actions/newGame";
import PlayerForm from "../forms/PlayerForm";
import { Link } from "react-router-dom";
import {
  FullSizeContainer,
  Button,
  FlexContainer,
  FlexSpacer,
  Paragraph
} from "../../components/helpers/Common";

const ButtonsContainer = styled.div`
  box-sizing: border-box;
  flex: 1;
`;

const NewGame = () => {
  return (
    <FullSizeContainer>
      <FlexContainer>
        <PlayerForm />
      </FlexContainer>
      <FlexContainer>
        <FlexSpacer />
        <ButtonsContainer>
          <Link to="create-party" id="create-party-link">
            <Button primary width="100%" marginTop="20px">
              CREATE A NEW PARTY
            </Button>
          </Link>
          <Paragraph gameFont color="accent" size="20px" center padding="20px">
            OR
          </Paragraph>
          <Link to="party-list" id="party-list-link">
            <Button primary width="100%">
              JOIN A PARTY
            </Button>
          </Link>
        </ButtonsContainer>
        <FlexSpacer />
      </FlexContainer>
    </FullSizeContainer>
  );
};

export default NewGame;
