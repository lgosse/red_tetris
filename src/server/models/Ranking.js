import mongoose from 'mongoose';

const rankingSchema = new mongoose.Schema({
  score: Number,
  playerName: String,
  date: Date
});

class Ranking {
  constructor(ranking) {
    this.date = new Date();
    this.score = ranking.score;
    playerName = ranking.playerName;
  }

  static findHighestRankings() {
    return this.find({})
      .sort('score')
      .limit(100);
  }
}

rankingSchema.loadClass(Ranking);
const RankingModel = mongoose.model('Ranking', rankingSchema);

export { rankingSchema, Ranking };

export default RankingModel;
