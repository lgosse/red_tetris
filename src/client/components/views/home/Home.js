import React from "react";
import { withRouter } from "react-router-dom";
import RedTetrisHeader from "./RedTetrisHeader";
import RedTetrisPreview from "./RedTetrisPreview";
import BeginAdventure from "./BeginAdventure";
import { FullSizeContainer, HexaSeparator } from "../../helpers/Common";

const Home = () => {
  return (
    <FullSizeContainer>
      <RedTetrisHeader />
      <HexaSeparator />
      <RedTetrisPreview />
      <BeginAdventure />
    </FullSizeContainer>
  );
};

export default Home;
