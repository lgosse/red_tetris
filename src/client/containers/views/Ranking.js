import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import globalStyle from '../../styles/global';

import { FullSizeContainer, Paragraph } from '../../components/helpers/Common';

const NoRanks = () => (
  <Paragraph
    gameFont
    size="20px"
    center
    style={{ color: globalStyle.color.accent }}
  >
    NO RANKINGS YET
  </Paragraph>
);

const Ranks = ({ rankings }) => <div />;

const Ranking = ({ ranking }) => (
  <FullSizeContainer padding="20px">
    {ranking.length ? <Ranks ranking={ranking} /> : <NoRanks />}
  </FullSizeContainer>
);

const mapStateToProps = ({ ranking }) => ({ ranking });

export default connect(mapStateToProps)(Ranking);
