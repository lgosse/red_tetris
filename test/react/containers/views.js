import chai from "chai";
import React from "react";
import equalJSX from "chai-equal-jsx";
import { createRenderer } from "react-dom/test-utils";

import { Connect, StyledComponent } from "../../helpers/mockComponents";

import Home from "../../../src/client/containers/views/Home";
import Ranking from "../../../src/client/containers/views/Ranking";
import NewGame from "../../../src/client/containers/views/NewGame";
import NotFound from "../../../src/client/containers/views/NotFound";
import {
  containerStyle,
  Code404,
  Message
} from "../../../src/client/containers/views/NotFound";

chai.should();
chai.use(equalJSX);

describe("Ranking", () => {
  it("should render as expected", () => {
    const renderer = createRenderer();
    renderer.render(React.createElement(Ranking));
    const output = renderer.getRenderOutput();
    output.should.matchSnapshot();
  });
});

describe("Home", () => {
  it("should render as expected", () => {
    const renderer = createRenderer();
    renderer.render(React.createElement(Home));
    const output = renderer.getRenderOutput();
    output.should.matchSnapshot();
  });
});

describe("NewGame", () => {
  it("should render as expected", () => {
    const renderer = createRenderer();
    renderer.render(React.createElement(NewGame));
    const output = renderer.getRenderOutput();
    output.should.matchSnapshot();
  });
});

describe("NotFound", () => {
  it("should render as expected", () => {
    const renderer = createRenderer();
    renderer.render(React.createElement(NotFound));
    const output = renderer.getRenderOutput();
    output.should.matchSnapshot();
  });
});
