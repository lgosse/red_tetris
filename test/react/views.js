import chai from "chai";
import React from "react";
import equalJSX from "chai-equal-jsx";
import { createRenderer } from "react-dom/test-utils";

import { Connect, StyledComponent } from "../helpers/mockComponents";

import Home from "../../src/client/components/views/home/Home";
import Ranking from "../../src/client/components/views/Ranking";
import NewGame from "../../src/client/components/views/new-game/NewGame";
import NotFound from "../../src/client/components/views/NotFound";
import {
  containerStyle,
  Code404,
  Message
} from "../../src/client/components/views/NotFound";

chai.should();
chai.use(equalJSX);

describe("Ranking", () => {
  it("should render as expected", () => {
    const renderer = createRenderer();
    renderer.render(React.createElement(Ranking));
    const output = renderer.getRenderOutput();
    output.should.equalJSX(<div>this is the ranking page</div>);
  });
});

describe("Home", () => {
  it("should render as expected", () => {
    const renderer = createRenderer();
    renderer.render(React.createElement(Home));
    const output = renderer.getRenderOutput();
    output.should.equalJSX(<div>this is a test</div>);
  });
});

describe("NewGame", () => {
  it("should render as expected", () => {
    const renderer = createRenderer();
    renderer.render(React.createElement(NewGame));
    const output = renderer.getRenderOutput();
    output.should.exist;
  });
});

describe("NotFound", () => {
  it("should render as expected", () => {
    const renderer = createRenderer();
    renderer.render(React.createElement(NotFound));
    const output = renderer.getRenderOutput();
    output.should.equalJSX(
      <div style={containerStyle}>
        <Code404>404</Code404>
        <Message>
          We're sorry but your requested page is unknown to us :(
        </Message>
      </div>
    );
  });
});
