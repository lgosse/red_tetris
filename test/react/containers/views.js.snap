// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render as expected 1`] = `
<styled.div>
  <RedTetrisHeader />
  <HexaSeparator />
  <RedTetrisPreview />
  <BeginAdventure />
</styled.div>
`;

exports[`NewGame should render as expected 1`] = `
<styled.div>
  <styled.div>
    <Connect(PlayerForm) />
  </styled.div>
  <styled.div>
    <styled.div />
    <styled.div>
      <styled.button
        marginTop="20px"
        primary={true}
        width="100%"
      >
        CREATE A NEW GAME
      </styled.button>
      <styled.button
        marginTop="20px"
        primary={true}
        width="100%"
      >
        JOIN A GAME
      </styled.button>
    </styled.div>
    <styled.div />
  </styled.div>
</styled.div>
`;

exports[`NotFound should render as expected 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <styled.div>
    404
  </styled.div>
  <styled.div>
    We're sorry but your requested page is unknown to us :(
  </styled.div>
</div>
`;

exports[`Ranking should render as expected 1`] = `
<div>
  this is the ranking page
</div>
`;
